lib/Connector/XiboAudienceReportingConnector.php

```diff
@@ -64,4 +111,195 @@ public function processSettingsForm(SanitizerInterface $params, array $settings)
         // TODO: Implement processSettingsForm() method.
         return $settings;
     }
+
+    /**
+     * @throws NotFoundException
+     * @throws GeneralException
+     */
+    public function onRegularMaintenance(MaintenanceRegularEvent $event)
+    {
+        $watermark = null;
+        // Get Watermark
+        try {
+            $response = $this->getClient()->get('https://test-exchange.xibo-adspace.com/api/audience/watermark', [
```

Comment:
I am not entirely sure how to override settings in `settings-custom.php`

Reply:
You can set "provider settings" for connectors using the following syntax:

```
$connectorSettings = [
    'xibo-ssp-connector' => [
        'serviceUrl' => 'https://test-exchange.xibo-adspace.com/api'
    ]
];
```

The array key is the connectors ID

When you're in the connector and call getSetting it will return you to provider setting, or the user setting if one has been set with the same key.

Reply:
Thanks, that is done.